File: src/main/java/org/jukeboxmc/event/Cancellable.java
package org.jukeboxmc.event;

/**
 * @author Kaooot
 * @version 1.0
 */
public interface Cancellable {

    /**
     * Updates the cancelled state to the given value
     *
     * @param cancelled which should be set
     */
    void setCancelled( boolean cancelled );

    /**
     * Retrieves whether the {@link Cancellable} implementation is cancelled or not
     *
     * @return whether the event implementation is cancelled
     */
    boolean isCancelled();
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/Cancellable.kt
package org.jukeboxmc.api.event

interface Cancellable {

    /**
     * Updates the cancelled state to the given value
     *
     * @param cancelled which should be set
     */
    fun setCancelled(cancelled: Boolean)

    /**
     * Retrieves whether the [Cancellable] implementation is cancelled or not
     *
     * @return whether the event implementation is cancelled
     */
    fun isCancelled(): Boolean
}